src/drivers/actuators/vertiq_io/vertiq_client_manager.cpp

```diff
@@ -0,0 +1,229 @@
+{
+	return _object_id_now;
+}
+
+IQUartFlightControllerInterfaceClient *VertiqClientManager::GetTelemIFCI(){
+	return _telem_ifci;
+}
+
+
+void VertiqClientManager::UpdateClientsToNewObjId(uint8_t new_object_id)
+{
+	_object_id_now = new_object_id;
+
+	delete _prop_input_parser_client;
+	_prop_input_parser_client = new EscPropellerInputParserClient(new_object_id);
+
+#ifdef CONFIG_USE_IFCI_CONFIGURATION
+	delete _ifci_client;
+	_ifci_client = new IQUartFlightControllerInterfaceClient(new_object_id);
+#endif
+
+#ifdef CONFIG_USE_PULSING_CONFIGURATION
+	delete _voltage_superposition_client;
+	_voltage_superposition_client = new VoltageSuperPositionClient(new_object_id);
+
+	delete _pulsing_rectangular_input_parser_client;
+	_pulsing_rectangular_input_parser_client = new PulsingRectangularInputParserClient(new_object_id);
+#endif
+}
+
+void VertiqClientManager::HandleClientCommunication()
+{
+	//Called periodically in the main loop to handle all communications not handled direclty by
+	//parameter setting
+	//Update our serial tx before we take in the rx
+	_serial_interface->process_serial_tx();
+
+	//Update our serial rx
+	_serial_interface->process_serial_rx(_configuration_client_array, _operational_client_array);
+}
+
+void VertiqClientManager::SendSetForceArm()
+{
+	_broadcast_arming_handler.motor_armed_.set(*_serial_interface->get_iquart_interface(), 1);
+}
+
+void VertiqClientManager::SendSetForceDisarm()
+{
+	_broadcast_arming_handler.motor_armed_.set(*_serial_interface->get_iquart_interface(), 0);
+}
+
+void VertiqClientManager::SendSetCoast()
+{
+	_broadcast_prop_motor_control.ctrl_coast_.set(*_serial_interface->get_iquart_interface());
+}
+
+void VertiqClientManager::SendSetVelocitySetpoint(uint16_t velocity_setpoint)
+{
+	_broadcast_prop_motor_control.ctrl_velocity_.set(*_serial_interface->get_iquart_interface(), velocity_setpoint);
+}
+
+void VertiqClientManager::MarkIquartConfigsForRefresh()
+{
+	for (uint8_t i = 0; i < _added_entry_wrappers; i++) {
+		_entry_wrappers[i]->SetNeedsInit();
+	}
+}
+
+void VertiqClientManager::UpdateIquartConfigParams()
+{
+	for (uint8_t i = 0; i < _added_entry_wrappers; i++) {
+		_entry_wrappers[i]->SendGet(_serial_interface);
+		//Ensure that these get messages get out
+		_serial_interface->process_serial_tx();
+	}
+
+	//Now go ahead and grab responses, and update everyone to be on the same page, but do it quickly.
+	CoordinateIquartWithPx4Params(100_ms);
+}
+
+void VertiqClientManager::CoordinateIquartWithPx4Params(hrt_abstime timeout)
+{
+	//Get the start time and the end time
+	hrt_abstime time_now = hrt_absolute_time();
+	hrt_abstime end_time = time_now + timeout;
+
+	while (time_now < end_time) {
+		for (uint8_t i = 0; i < _added_entry_wrappers; i++) {
+			_entry_wrappers[i]->Update(_serial_interface);
+		}
+
+		//Update the time
+		time_now = hrt_absolute_time();
+
+		//Update our serial rx
+		_serial_interface->process_serial_rx(_configuration_client_array, _operational_client_array);
```

Comment:
Do you need the operational guys here? Or do you only need configuration ones here?

Reply:
Need both. Everyone, regardless of who you are needs to see if there's a response there for you. For example, in the demo, there are the 2 multi turn clients. They need to be able to check if someone responded with their current angle.